### DISCUSSION TITLE
Property '#private' is missing in type 'Config<PublicClient<FallbackTransport>, WebSocketPublicClient<FallbackTransport>> & { ...; }' but required in type 'Config<any, any>'.

### DISCUSSION THREAD

**Discussion by victorpavlenko** [WalletConnect/walletconnect-monorepo#3819] [Developer Support]
```
 const { chains, publicClient, webSocketPublicClient } =
        configureChains([mainnet, bsc, polygon], [
          publicProvider(),
        ]);

      const connectors: any = [
        new WalletConnectConnector({ chains, options: { projectId, showQrModal: false } }),
        new EIP6963Connector({ chains }),
        new CoinbaseWalletConnector({ chains, options: { appName: metadata.name } })
      ];

      const wagmiConfig = createConfig({
        autoConnect: false,
        connectors,
        publicClient,
        webSocketPublicClient
      });

      console.log(wagmiConfig);
      console.log(wagmiConfig);

      createWeb3Modal({
        wagmiConfig,
        projectId,
        chains
      });

```

v3.1.0

I have no idea what does it mean but my TS build doesnt work is it related to wagmi to web3 or TS?

**Comment by glitch-txs** [comment]
Could you provide more context?
- which framework are you using?
- steps to reproduce?
- do you have a MRE?

**Comment by Jsherborne92** [comment]
Same issue here.... using react.... the solution for me was to only install wagmi and not also install @wagmi/core.

**Comment by mwawrusch** [comment]
Ran into this as well. Having used wagmi on many apps over the last year I have to say that these kind of issues pop up regularly and are a huge productivity killer. It seems to normally be version conflicts in some of the underlying libraries, something that would be great if it could be addressed. Thinking of some validation suite that ensures that the dependencies are correct.

**Comment by mwawrusch** [comment]
To add to that, the generally 'best' solution is to ensure that you pin the dependencies exactly, install the latest of all the web3 dependent libraries, like wagmi, viem, aa-, zerodev, magic, etc and pray!!! Then wipe out your node_modules, delete your package-lock.json and reinstall. 

Finally, use a monorepo like turbo to simplify dependency management.

**Comment by 124maa** [comment]
> ```
>  const { chains, publicClient, webSocketPublicClient } =
>         configureChains([mainnet, bsc, polygon], [
>           publicProvider(),
>         ]);
> 
>       const connectors: any = [
>         new WalletConnectConnector({ chains, options: { projectId, showQrModal: false } }),
>         new EIP6963Connector({ chains }),
>         new CoinbaseWalletConnector({ chains, options: { appName: metadata.name } })
>       ];
> 
>       const wagmiConfig = createConfig({
>         autoConnect: false,
>         connectors,
>         publicClient,
>         webSocketPublicClient
>       });
> 
>       console.log(wagmiConfig);
>       console.log(wagmiConfig);
> 
>       createWeb3Modal({
>         wagmiConfig,
>         projectId,
>         chains
>       });
> 
> ```
> 
> v3.1.0
> 
> I have no idea what does it mean but my TS build doesnt work is it related to wagmi to web3 or TS?

**Comment by tpae** [comment]
I'm getting this as well. Is it related to versions?

**Comment by tpae** [comment]
I solved this by making sure `@wagmi/core` was the same version as `wagmi` This happened after trying to install older version of `@web3modal/wagmi`.

```
    "@wagmi/core": "1.4.3", <-- examples don't show this, but make sure you use the same version as your wagmi
    "@web3modal/wagmi": "3.0.2",
    "viem": "1.14.0",
    "wagmi": "1.4.3",
```

**Comment by 124maa** [comment]
> ```
>  const { chains, publicClient, webSocketPublicClient } =
>         configureChains([mainnet, bsc, polygon], [
>           publicProvider(),
>         ]);
> 
>       const connectors: any = [
>         new WalletConnectConnector({ chains, options: { projectId, showQrModal: false } }),
>         new EIP6963Connector({ chains }),
>         new CoinbaseWalletConnector({ chains, options: { appName: metadata.name } })
>       ];
> 
>       const wagmiConfig = createConfig({
>         autoConnect: false,
>         connectors,
>         publicClient,
>         webSocketPublicClient
>       });
> 
>       console.log(wagmiConfig);
>       console.log(wagmiConfig);
> 
>       createWeb3Modal({
>         wagmiConfig,
>         projectId,
>         chains
>       });
> 
> ```
> 
> v3.1.0
> 
> I have no idea what does it mean but my TS build doesnt work is it related to wagmi to web3 or TS?
